### PR TITLE
Pr snapdragon follow up

### PR DESCRIPTION
**Pull request by RomanBapst**
Replaces #9467 

@dagar I cherry-picked the commits from my previous PR which were not addressed by your recent PR's.

### CODE REVIEW

**Comment by dagar** [comment]
Looks good. I don't have an answer for putchar at the moment.

File: src/platforms/posix/drivers/df_mpu9250_wrapper/df_mpu9250_wrapper.cpp [hunk 1]
```diff
@@ -165,7 +165,7 @@ class DfMpu9250Wrapper : public MPU9250
 	math::LowPassFilter2p	_gyro_filter_y;
 	math::LowPassFilter2p	_gyro_filter_z;
 
-	unsigned		    _publish_count;
+	//unsigned		    _publish_count;
```
Comment (RomanBapst) [inline src/platforms/posix/drivers/df_mpu9250_wrapper/df_mpu9250_wrapper.cpp]: Note to myself: Remove this

**Comment by dagar** [comment]
Driverframework was on an invalid commit after the PR was merged. I updated the submodule and squashed into the previous commit.

File: platforms/qurt/src/px4_layer/px4_qurt_impl.cpp [hunk 2]
```diff
@@ -166,3 +166,9 @@ int fputc(int c, FILE *stream)
 {
 	return c;
 }
+
+int putchar(int character)
+{
+	PX4_ERR("Error: Calling unresolved symbol stub:[%s]", __FUNCTION__);
```
Comment (RomanBapst) [inline platforms/qurt/src/px4_layer/px4_qurt_impl.cpp]: @dagar Is is better if we just print nothing for now? The only thing I worry about is that we forget about this and at some point it bites us.

**Comment by RomanBapst** [comment]
@dagar @bkueng Can somebody review this so that we can bring it in.
Not sure what to do about the putchar. One suggestion would be to just make an empty function stub. I assume it's just used for printing to the console so at least it's nothing critical.

**Comment by dagar** [comment]
@RomanBapst things seem to be working and we can come back to it later if we ever develop a better understanding.

**Pull request merged**